### DISCUSSION TITLE
Add TabItem from code: not inheriting size

### DISCUSSION THREAD

**Discussion by le-rou** [AvaloniaUI/Avalonia#15128] [General]
Hi, I'm trying to add TabItems to a TabControl from code as I've done in WPF before. I can add the item just fine, but the TabItem doesn't scale - it doesn't seem to get the parent's size. Is there something more I need to do? Adding the TabItem with the same content in the AXAML works.

```
// create a tab Item, bind the header to the ViewModel
TabItem tabItem = new TabItem();
tabItem.DataContext = areaVM;
tabItem.Bind(TabItem.HeaderProperty, new Binding(nameof(areaVM.Name)));

// create a user control, bind it to the content
var areaView = new AreaView();
areaView.DataContext = areaVM;
tabItem.Content = areaView;

// add the tab item to the tab control
AreaContainer.Items.Add( tabItem );
AreaContainer.SelectedItem = tabItem;
```

The content is set to a user control. The user control contains a rectangle set to stretch in both directions, so it should adjust to its parent's size. When running the application, I can add the TabItem to the TabControl and it shows up. However, it's height seems to be 100%, but the width is limited to the TabItem's header's width for some reason. I would really expect the width to be 100% of the parent as well - the parent being the TabControl that uses the full window's width.

Can anyone give me a hint? Thanks in advance!

![image](https://github.com/AvaloniaUI/Avalonia/assets/20772173/af51f60c-a811-4e88-8b38-9faadc2c7efd)

**Comment by timunie** [comment]
Check Alignments in DevTools. Make sure TabControl is not added into a StackPanel.

**Reply by le-rou** [reply]
I restructured the AXAML that contains the TabControl and now it works. Thank you for the quick response, saving my day.
